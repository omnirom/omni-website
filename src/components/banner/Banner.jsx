import './_banner.scss';
import React, {Component} from 'react';
import PropTypes from 'prop-types';

export class Banner extends Component {
  render() {
    const {
      bgImageUrl,
      centerImageUrl,
      heading,
      subHeading
    } = this.props;
    const imageStyle = {
      "backgroundImage": `linear-gradient(180deg, rgba(0,0,0,0.2), rgba(0,0,0,0.7) 80%, #000), url(${bgImageUrl})`
    };

    return (
      <div className="banner" style={imageStyle}>
        <div className="banner__image">
          <img src={centerImageUrl} />
        </div>
        <h1 className="banner__heading">{heading}</h1>
        <p className="banner__sub-heading">{subHeading}</p>
      </div>
    );
  }
}

Banner.propTypes = {
  bgImageUrl: PropTypes.string,
  centerImageUrl: PropTypes.string,
  heading: PropTypes.string,
  subHeading: PropTypes.string
};

export default Banner;
