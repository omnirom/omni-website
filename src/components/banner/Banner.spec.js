import React from 'react';
import {Banner} from './Banner.jsx';
import renderer from 'react-test-renderer';

describe('Banner Component', () => {
  it('renders banner', () => {
    const component = renderer.create(
      <Banner
        bgImageUrl="bg-image.png"
        centerImageUrl="center-image.png"
        heading="Banner Title"
        subHeading="additional sub text goes here"
      />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
