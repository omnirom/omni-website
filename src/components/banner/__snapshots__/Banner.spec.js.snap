// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Component renders banner 1`] = `
<div
  className="banner"
  style={
    Object {
      "backgroundImage": "linear-gradient(180deg, rgba(0,0,0,0.2), rgba(0,0,0,0.7) 80%, #000), url(bg-image.png)",
    }
  }
>
  <div
    className="banner__image"
  >
    <img
      src="center-image.png"
    />
  </div>
  <h1
    className="banner__heading"
  >
    Banner Title
  </h1>
  <p
    className="banner__sub-heading"
  >
    additional sub text goes here
  </p>
</div>
`;
